src/Input/InputEventsActions.cpp

```diff
@@ -334,12 +336,36 @@ InputEvents::eventStatus(const TCHAR *misc)
 void
 InputEvents::eventAnalysis([[maybe_unused]] const TCHAR *misc)
 {
+
+  const std::map<std::string_view, AnalysisPage> stringToAnalysisPageEnum{
```

Comment:
Trying to be clever, yes but not really ; Actually I found this solution more elegant than the other one...
I switch to my first solution, which make my eyes bleeding but is far more efficient :
```cpp
AnalysisPage analysis_page;

if (StringIsEqual(misc, _T("AnalysisPage::BAROGRAPH"))) {
analysis_page = AnalysisPage::BAROGRAPH;
} else if (StringIsEqual(misc, _T("AnalysisPage::COUNT"))) {
analysis_page = AnalysisPage::COUNT;
} else if (StringIsEqual(misc, _T("AnalysisPage::CLIMB"))) {
analysis_page = AnalysisPage::CLIMB;
} else if (StringIsEqual(misc, _T("AnalysisPage::VARIO_HISTOGRAM"))) {
analysis_page = AnalysisPage::VARIO_HISTOGRAM;
} else if (StringIsEqual(misc, _T("AnalysisPage::THERMAL_BAND"))) {
analysis_page = AnalysisPage::THERMAL_BAND;
} else if (StringIsEqual(misc, _T("AnalysisPage::WIND"))) {
analysis_page = AnalysisPage::WIND;
} else if (StringIsEqual(misc, _T("AnalysisPage::POLAR"))) {
analysis_page = AnalysisPage::POLAR;
} else if (StringIsEqual(misc, _T("AnalysisPage::MACCREADY"))) {
analysis_page = AnalysisPage::MACCREADY;
} else if (StringIsEqual(misc, _T("AnalysisPage::TEMPTRACE"))) {
analysis_page = AnalysisPage::TEMPTRACE;
} else if (StringIsEqual(misc, _T("AnalysisPage::TASK"))) {
analysis_page = AnalysisPage::TASK;
} else if (StringIsEqual(misc, _T("AnalysisPage::CONTEST"))) {
analysis_page = AnalysisPage::CONTEST;
} else if (StringIsEqual(misc, _T("AnalysisPage::TASK_SPEED"))) {
analysis_page = AnalysisPage::TASK_SPEED;
} else if (StringIsEqual(misc, _T("AnalysisPage::AIRSPACE"))) {
analysis_page = AnalysisPage::AIRSPACE;
} else {
analysis_page = AnalysisPage::CONTEST;
}

// instead of:

const std::map<std::string_view, AnalysisPage> stringToAnalysisPageEnum{
{"AnalysisPage::BAROGRAPH", AnalysisPage::BAROGRAPH},
{"AnalysisPage::COUNT", AnalysisPage::COUNT},
{"AnalysisPage::CLIMB", AnalysisPage::CLIMB},
{"AnalysisPage::VARIO_HISTOGRAM", AnalysisPage::VARIO_HISTOGRAM},
{"AnalysisPage::THERMAL_BAND", AnalysisPage::THERMAL_BAND},
{"AnalysisPage::WIND", AnalysisPage::WIND},
{"AnalysisPage::POLAR", AnalysisPage::POLAR},
{"AnalysisPage::MACCREADY", AnalysisPage::MACCREADY},
{"AnalysisPage::TEMPTRACE", AnalysisPage::TEMPTRACE},
{"AnalysisPage::TASK", AnalysisPage::TASK},
{"AnalysisPage::CONTEST", AnalysisPage::CONTEST},
{"AnalysisPage::TASK_SPEED", AnalysisPage::TASK_SPEED},
{"AnalysisPage::AIRSPACE", AnalysisPage::AIRSPACE}};

AnalysisPage APage;
if (stringToAnalysisPageEnum.contains(misc))
APage = stringToAnalysisPageEnum.at(misc);
else
APage = stringToAnalysisPageEnum.at("AnalysisPage::CONTEST");

```
is it OK for you ?

for info, in the next coming year, it will be possible to use this in the standard library. I could then try to be clever again ;) :
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2996r0.html#enum-to-string

other ideas:
https://github.com/Neargye/magic_enum